File: app/src/main/java/com/hoony/dagger2simpleexample3/chaper3/Named.java
package com.hoony.dagger2simpleexample3.chaper3;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import javax.inject.Qualifier;

@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
public @interface Named {
    String value() default "";
}

File: app/src/main/java/com/hoony/dagger2simpleexample3/chaper3/Named.kt
package com.hoony.dagger2simpleexample3.chaper3

import javax.inject.Qualifier

@Qualifier
@MustBeDocumented
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class Named(val value: Name = Name.LOVE) {
    enum class Name {
        LOVE, HELLO
    }
}